File: ProductApi/ProductApi/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace ProductApi.Entities
{
    public class Product
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        public string Name { get; set; }
        public string? Description { get; set; }

        public string Sku { get; set; }

        public decimal BasePrice { get; set; }

        public Guid CategoryId { get; set; }
        public Category? Category { get; set; }

        public bool IsActive { get; set; } = true;

        public List<ProductVariant> Variants { get; set; } = new();
        public List<ProductImage> Images { get; set; } = new();

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [Timestamp]
        public byte[]? RowVersion { get; set; }
    }

}

File: FE/frontend/src/types/product.ts
export interface ProductImage {
  id: string;
  productId: string;
  imageUrls: string;
  isPrimary: boolean;
}

export interface ProductVariant {
  id: string | null;     
  productId?: string;
  color?: string;
  size?: string;
  additionalPrice: number;
  stockQuantity: number;
}

export interface Product {
  id: string;
  name: string;
  description?: string;
  sku: string;
  basePrice: number;
  categoryId: string;
  isActive: boolean;
  variants: ProductVariant[];
  imageUrls: string[];   
  createdAt: string;
  updatedAt: string;
  rowVersion: string;
}


export interface CreateProductRequest {
  name: string;
  description?: string;
  sku: string;
  basePrice: number;
  categoryId: string;
  variants: ProductVariant[];
  imageUrls: string[];
}

export interface UpdateProductRequest {
  name: string;
  description?: string;
  basePrice: number;
  categoryId: string;
  rowVersion: string;              
  variants: ProductVariant[];
  imageUrls: string[];
}
